third_party/lwip/repo/lwip/src/core/dns.c

```diff
@@ -1518,6 +1518,7 @@ dns_recv(void *arg, struct udp_pcb *pcb, struct pbuf *p, const ip_addr_t *addr,
             entry->state = DNS_STATE_ASKING;
           } else {
             dns_call_found(i, NULL);
+            dns_table[i].state = DNS_STATE_UNUSED;
```

Comment:
There are a number of other places where we see a similar pattern, i.e. we make a call to `dns_call_found(i, NULL)` without changing the state of the `dns_table`.  Most similar is in line 1398 above, but also in 1380.  Are those code paths subject to the same issue, and should they result in cleaning up the state of the DNS table?